media-sound/lilypond/lilypond-2.22.2.recipe

```diff
@@ -0,0 +1,183 @@
+	"
+ARCHITECTURES_doc="?any"
+
+languages="ca cs de es fr hu it ja nl pt zh"
+for lang in $languages; do
+	eval "PROVIDES_doc_${lang}=\"\
+		lilypond${secondaryArchSuffix}_doc_${lang} = $portVersion\
+		\"; \
+	REQUIRES_doc_${lang}=\"\
+		lilypond${secondaryArchSuffix}_doc == $portVersion base\
+		\"; \
+	SUMMARY_doc_${lang}=\"${SUMMARY} (documentation ${lang})\"\
+	ARCHITECTURES_doc_${lang}=\"?any\""
+done
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	tex_gyre
+	texlive_basic
+	# most of these texlive_lang* packages are needed by the tex engines
+	texlive_langchinese
+	texlive_langcyrillic # this one is specifically needed by lilypond
+	texlive_langczechslovak
+	texlive_langenglish
+	texlive_langeuropean
+	texlive_langfrench
+	texlive_langgerman
+	texlive_langgreek
+	texlive_langitalian
+	texlive_langother
+	texlive_langpolish
+	texlive_langportuguese
+	texlive_langspanish
+	texlive_metapost
+	texlive_plaingeneric
+	urw_base35_fonts
+	cmd:guile < 2
+	devel:libfreetype$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libgobject_2.0$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix
+	devel:libgs$secondaryArchSuffix
+	devel:libharfbuzz$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	devel:libpangoft2_1.0$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+#	cmd:awk
+	cmd:bison
+	cmd:convert$secondaryArchSuffix
+	cmd:dblatex
+	cmd:extractpdfmark
+	cmd:fc_list$secondaryArchSuffix
+	cmd:find
+	cmd:flex
+	cmd:fontforge
+	cmd:gcc$secondaryArchSuffix
+	cmd:kpsewhich
+	cmd:make
+	cmd:mawk # required by texindex
+	cmd:mf
+	cmd:mpost
+	cmd:perl
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:rsync
+	cmd:t1asm
+	cmd:tar
+	cmd:texi2html
+	cmd:texindex
+	cmd:tidy
+	cmd:xelatex
+	cmd:zip
+	"
+
+TEST_REQUIRES="
+	# either one of these is required (don't know which yet, it doesn't work yet)
+#	dejavu
+#	texlive_fontsextra
+	texlive_xetex
+	cmd:git
+	cmd:xsltproc
+	"
+
+defineDebugInfoPackage lilypond$secondaryArchSuffix \
+	$prefix/bin/lilypond
+
+PATCH()
+{
+	./autogen.sh --noconfigure
+}
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+
+	# the build system already installs documentation in a subdirectory of docdir
+	runConfigure --omit-dirs "binDir docDir" $sourceDir/configure \
+		--bindir=$prefix/bin \
+		--docdir=$documentationDir/packages \
+		--with-localedir=$dataDir/locale \
+		--enable-gs-api # needed because of broken ghostscript (gs)
+	make $jobArgs
+
+	# CPU_COUNT=$jobs is not used because it causes hangs
+	# strip the current directory (.) from PATH, it disallows find -execdir
+	PATH=${PATH:2} make $jobArgs doc
+}
+
+INSTALL()
+{
+	cd build
+	make install
+	PATH=${PATH:2} make install-doc
+
+	# language specific doc packages
+	for lang in $languages; do
+		packageEntries doc_${lang} \
+			$(find $docDir -name "*.${lang}.*")
+	done
+
+	# generic and english doc package
+	packageEntries doc \
+		$docDir $infoDir
+}
+
+TEST()
+{
```

Comment:
should I add `fc-cache -f` here to make sure?

Reply:
And in TEST_REQUIRES also, could you run the tests with/without the urw fonts? (I'm guessing it searches for the gostscript fonts?)

Reply:
the URW fonts are in BUILD_REQUIRES anyway. I think TEST_REQUIRES just contains the additional requirements for tests?

Reply:
Yes, like fc-cache isn't needed at runtime nor build time, so only needs adding in TEST_REQUIRES :) (and as you mentioned it above in TEST() (PS, not goind to build this monster again to check, so I take your word for it) :)